src/pages/EmailPage/container.jsx

```diff
@@ -1,33 +1,98 @@
 import React from 'react';
+import queryString from 'query-string'
+import { NotificationManager } from 'react-notifications';
+import { navigate } from '@reach/router';
 
 import EmailPageView from './view';
+import ApiService from '../../services/apiService';
 
 class EmailPage extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      email: '',
-      text: '',
+      emails: '',
+      title: '',
+      message: '',
+      isLoading: false,
+      isSuccessfullySubmitted: false,
     };
   }
 
+  componentDidMount() {
+    const queryParams = queryString.parse(this.props.location.search);
+    const ids = queryParams.ids;
+    if (ids){
+      this.fetchData(ids);
+    }
+  }
+
+  fetchData = (ids) => {
+    ApiService.call({
+      method: 'get',
+      url: `/contacts/email?id=${ids}`,
+    })
+      .then((response) => {
+        const emails = response.data.map(record => record.email);
+        this.setState({ emails: emails.join(',') });
+      })
+      .catch((error) => {
+        console.log(error);
+        navigate('/contacts');
+      });
+  };
+
   handleChange = (event) => {
     const { name, value } = event.target;
     this.setState({ [name]: value });
   };
 
-  handleSubmit = (event) => {
-    console.log('submit');
+  handleSubmit = () => {
+    const { isValid } = this.validate();
+
+    if (isValid) {
+      this.submit();
+    }
+  };
+
+  validate = () => {
+    return {
+      errors: [],
+      isValid: true,
+    };
+  };
+
+  submit = () => {
+    const { emails, title, message } = this.state;
+    this.setState({ isLoading: true });
+    ApiService.call({
+      method: 'post',
+      url: '/contacts/email',
+      data: { emails: emails.split(','), title, message }
+    }, {
+      isAuthorizedRequest: true
```

Comment:
You can omit second argument of `ApiService.call`, because by default it is[ `{ isAuthorizedRequest: true }`](https://github.com/KatyaBaranovskaya/contact-directory-fe/blob/dev/src/services/apiService.js#L34).